internal/mode/static/state/graph/policies.go

```diff
@@ -0,0 +1,310 @@
+)
+
+// Policy represents an NGF Policy.
+type Policy struct {
+	// Source is the corresponding Policy resource.
+	Source policies.Policy
+	// Ancestor is the ancestor object of the Policy. Used in status.
+	Ancestor *PolicyAncestor
+	// TargetRef is the resource that the Policy targets.
+	TargetRef PolicyTargetRef
+	// Conditions holds the conditions for the Policy.
+	// These conditions apply to the entire Policy.
+	// The conditions in the Ancestor apply only to the Policy in regard to the Ancestor.
+	Conditions []conditions.Condition
+	// Valid indicates whether the Policy is valid.
+	Valid bool
+}
+
+// PolicyAncestor represents an ancestor of a Policy.
+type PolicyAncestor struct {
+	// Ancestor is the ancestor object.
+	Ancestor v1alpha2.ParentReference
+	// Conditions contains the list of conditions of the Policy in relation to the ancestor.
+	Conditions []conditions.Condition
+}
+
+// PolicyAncestorRef contains the identifying information of the ancestor.
+type PolicyAncestorRef struct {
+	// Nsname is the NamespacedName of the Ancestor.
+	Nsname types.NamespacedName
+	// Kind is the Kind of the Ancestor.
+	Kind v1.Kind
+	// Group is the Group of the Ancestor.
+	Group v1.Group
+}
+
+// PolicyTargetRef represents the object that the Policy is targeting.
+type PolicyTargetRef struct {
+	// Kind is the Kind of the object.
+	Kind v1.Kind
+	// Group is the Group of the object.
+	Group v1.Group
+	// Nsname is the NamespacedName of the object.
+	Nsname types.NamespacedName
+}
+
+// PolicyKey is a unique identifier for an NGF Policy.
+type PolicyKey struct {
+	// Nsname is the NamespacedName of the Policy.
+	NsName types.NamespacedName
+	// GVK is the GroupVersionKind of the Policy.
+	GVK schema.GroupVersionKind
+}
+
+const (
+	gatewayGroupKind = v1.GroupName + "/" + kinds.Gateway
+	hrGroupKind      = v1.GroupName + "/" + kinds.HTTPRoute
+	grpcGroupKind    = v1.GroupName + "/" + kinds.GRPCRoute
+)
+
+// attachPolicies attaches the graph's processed policies to the resources they target. It modifies the graph in place.
+func (g *Graph) attachPolicies(ctlrName string) {
+	if g.Gateway == nil {
+		return
+	}
+
+	for _, policy := range g.NGFPolicies {
+		ref := policy.TargetRef
+
+		switch ref.Kind {
+		case kinds.Gateway:
+			attachPolicyToGateway(policy, g.Gateway, g.IgnoredGateways, ctlrName)
+		case kinds.HTTPRoute, kinds.GRPCRoute:
+			route, exists := g.Routes[routeKeyForKind(ref.Kind, ref.Nsname)]
+			if !exists {
+				continue
+			}
+
+			attachPolicyToRoute(policy, route, ctlrName)
+		}
+	}
+}
+
+func attachPolicyToRoute(policy *Policy, route *L7Route, ctlrName string) {
+	kind := v1.Kind(kinds.HTTPRoute)
+	if route.RouteType == RouteTypeGRPC {
+		kind = kinds.GRPCRoute
+	}
+
+	routeNsName := types.NamespacedName{Namespace: route.Source.GetNamespace(), Name: route.Source.GetName()}
+
+	ancestor := &PolicyAncestor{
+		Ancestor: createParentReference(v1.GroupName, kind, routeNsName),
```

Comment:
After much thought and experimentation, I decided to use the target object as the ancestor object (Gateway when attached to a Gateway, Route when attached to a Route) because it is the simplest implementation and still provides the user enough information to piece together the entire status picture.

We can always change this in the future if we get feedback that it isn't helpful.

Reply:
I'm adding multiple target ref support in my work on the ObservabilityPolicy. For this policy, the targetRefs are also the ancestors, so it's still a 1:1 mapping. At some point though we may have to refactor that if those two things diverge.